src/linker/utilities/singularity_utils.py

```diff
@@ -1,66 +0,0 @@
-import os
-import subprocess
-from pathlib import Path
-from typing import Dict, List, Optional
-
-from loguru import logger
-
-
-def run_with_singularity(
-    input_data: List[Path],
-    results_dir: Path,
-    diagnostics_dir: Path,
-    step_id: str,
-    container_path: Path,
-    implementation_config: Optional[Dict[str, str]],
-) -> None:
-    logger.info(f"Running step {step_id} container with singularity")
-    cmd = _build_cmd(input_data, results_dir, diagnostics_dir, container_path)
-    _run_cmd(cmd, diagnostics_dir, implementation_config)
-    _clean(results_dir, container_path)
-
-
-def _build_cmd(
-    input_data: List[Path], results_dir: Path, diagnostics_dir: Path, container_path: Path
-) -> str:
-    cmd = (
-        "singularity run --containall --no-home --bind /tmp:/tmp "
```

Comment:
actually, some of these utils might come back in some form, since it seems we will need to pass additional arguments to singularity.